src/OrchardCore.Modules/OrchardCore.Search.Elasticsearch/Permissions.cs

```diff
@@ -7,8 +8,10 @@ namespace OrchardCore.Search.Elasticsearch;
 
 public class Permissions : IPermissionProvider
 {
+    [Obsolete("This property will be removed in future release. Instead use 'ElasticsearchIndexPermissionHelper.ManageElasticIndexes'.")]
```

Comment:
In this case, I don't think this is applicable. Using `ElasticsearchIndexPermissionHelper` is not better (since it's in the same project), but rather worse, since now you have two classes containing Elasticsearch permissions, and one of them is in a `Helper` class, which is a common name for static utility classes. It's not referenced anywhere else either.

So instead, keep this as the permission to rely on publicly and make `ElasticsearchIndexPermissionHelper.ManageElasticIndexes` internal.

Reply:
Okay. I obsoleted the instance in `ElasticsearchIndexPermissionHelper` and pointed it to `Permissions.ManageElasticIndexes`